src/frontend/src/handler/create_mv.rs

```diff
@@ -63,19 +67,41 @@ pub fn gen_create_mv_plan(
     let mut table = materialize.table().to_prost(schema_id, database_id);
     let plan: PlanRef = materialize.into();
     table.owner = session.user_name().to_string();
+    table.properties = properties;
 
     Ok((plan, table))
 }
 
+fn handle_create_mv_with_properties(options: Vec<SqlOption>) -> Result<HashMap<String, String>> {
+    // TODO: distinguish x.name
+    options
+        .into_iter()
+        .map(|x| match x.value {
+            Value::SingleQuotedString(s) => Ok((x.name.value, s)),
+            Value::Number(n, _) => Ok((x.name.value, n)),
+            _ => Err(RwError::from(ProtocolError(
+                "create_mv with properties only support single quoted string value".to_string(),
+            ))),
+        })
+        .collect()
+}
+
 pub async fn handle_create_mv(
     context: OptimizerContext,
     name: ObjectName,
     query: Box<Query>,
+    with_options: WithProperties,
 ) -> Result<PgResponse> {
     let session = context.session_ctx.clone();
 
     let (table, graph) = {
-        let (plan, table) = gen_create_mv_plan(&session, context.into(), query, name)?;
+        let (plan, table) = gen_create_mv_plan(
+            &session,
+            context.into(),
+            query,
+            name,
+            handle_create_mv_with_properties(with_options.0)?,
```

Comment:
Why not simply done this inside `gen_create_mv_plan` so we can avoid change the function signature